pkg/daemon/daemon.go

```diff
@@ -469,6 +471,28 @@ func (dn *Daemon) runOnceFrom() error {
 	return errors.New("unsupported onceFrom type provided")
 }
 
+// InstallSignalHandler installs the handler for the signals the daemon should act on
+func (dn *Daemon) InstallSignalHandler(stopCh chan struct{}) {
+	termChan := make(chan os.Signal, 2048)
+	signal.Notify(termChan, syscall.SIGTERM)
+
+	// Catch SIGTERM - if we're actively updating, we should avoid
+	// having the process be killed.
+	// https://github.com/openshift/machine-config-operator/issues/407
+	go func() {
+		for sig := range termChan {
+			switch sig {
+			case syscall.SIGTERM:
+				if dn.installedSigterm {
+					glog.Info("Got SIGTERM, but actively updating")
+				} else {
+					close(stopCh)
```

Comment:
Copying from the BZ your comment:
> Now, in the MCD we have an handler for sigterm only during our sync, the rest of the code doesn't really care about sigterm so we don't catch it and we exit with 143 instead of 0 (if we had an handler).

I have serious difficulty believing that all other pods have `SIGTERM` handlers of this form.  Being killed by `SIGTERM` seems like a *completely normal* event - what makes the MCD special here?  

Maybe it's that most other pods are drained rather than killed by kubelet?  If that's the case...OK I guess, but I would vehemently argue that whatever is treating "container got `SIGTERM`" as an error is the real problem.

Reply:
other operators aren't indeed getting sigtermed like us (we're a daemonset and we don't get drained).



> but I would vehemently argue that whatever is treating "container got `SIGTERM`" as an error is the real problem.

I would be with you :)